### PR TITLE
Convert InviteDialog to TypeScript

### PR DESCRIPTION
**Pull request by dbkr**
Before I start using it to select VoIP transfer targets

### CODE REVIEW

File: src/components/views/dialogs/InviteDialog.tsx [hunk 1]
```diff
@@ -298,28 +299,45 @@ class DMRoomTile extends React.PureComponent {
...
+
+interface IInviteDialogState {
+    targets: RoomMember[]; // array of Member objects (see interface above)
+    filterText: string;
+    recents: { user: DirectoryMember, userId: string }[];
+    numRecentsShown: number;
+    suggestions: { user: DirectoryMember, userId: string }[];
+    numSuggestionsShown: number;
+    serverResultsMixin: { user: DirectoryMember, userId: string }[];
+    threepidResultsMixin: ({ user: ThreepidMember, userId: string} | { user: DirectoryMember, userId: string})[];
```
Comment (turt2live) [inline src/components/views/dialogs/InviteDialog.tsx:325]: in theory these and all the other `DirectoryMember` definitions of the state can just use `Member` as the base class - there shouldn't be a need for them to use the specific types.

File: src/components/views/dialogs/InviteDialog.tsx [hunk 2]
```diff
@@ -603,7 +621,7 @@ export default class InviteDialog extends React.PureComponent {
             return;
         }
 
-        const createRoomOptions = {inlineErrors: true};
+        const createRoomOptions = {inlineErrors: true} as any;
```
Comment (turt2live) [inline src/components/views/dialogs/InviteDialog.tsx:624]:
```suggestion
        const createRoomOptions = {inlineErrors: true} as any; // XXX: Type out `createRoomOptions`
```

**Review by turt2live — APPROVED** [review]
lgtm with just the couple points

File: src/components/views/dialogs/InviteDialog.tsx (hunk 1 above)
Comment (dbkr) [reply src/components/views/dialogs/InviteDialog.tsx]: ah yep, that makes it simpler

**Pull request merged**